manager/logbroker/broker.go

```diff
@@ -383,6 +383,15 @@ func (lb *LogBroker) PublishLogs(stream api.LogBroker_PublishLogsServer) (err er
 			}
 		}
 
+		// if we have a close message, close out the subscription
+		if log.Close {
+			// Mark done and then set to nil so if we error after this point,
+			// we don't try to close again in the defer
+			currentSubscription.Done(remote.NodeID, err)
+			currentSubscription = nil
+			return stream.SendAndClose(&api.PublishLogsResponse{})
```

Comment:
Unless there is a race happening, I believe calling `CloseSend` doesn't preclude `SendAndClose`. In my testing, calling `SendAndClose` doesn't cause an error. 

On the other hand, I have no idea if that's really true, and even if it was, what purpose would returning an error here serve? If some errors happen after the part we care about has successfully completed, do we even have to care? 

So yeah i just changed that to `return nil`.